s3_test.go

```diff
@@ -0,0 +1,107 @@
+package veneur
+
+import (
+	"encoding/json"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/stretchr/testify/assert"
+)
+
+const S3TestBucket = "stripe-test-veneur"
+
+type mockS3Client struct {
+	s3iface.S3API
+	putObject func(*s3.PutObjectInput) (*s3.PutObjectOutput, error)
+}
+
+func (m *mockS3Client) PutObject(input *s3.PutObjectInput) (*s3.PutObjectOutput, error) {
+	return m.putObject(input)
+}
+
+// TestS3Post tests that we can correctly post a sequence of
+// DDMetrics to S3
+func TestS3Post(t *testing.T) {
+	RemoteResponseChan := make(chan struct{}, 1)
+	defer func() {
+		select {
+		case <-RemoteResponseChan:
+			// all is safe
+			return
+		case <-time.After(DefaultServerTimeout):
+			assert.Fail(t, "Global server did not complete all responses before test terminated!")
+		}
+	}()
```

Comment:
random thought: we've got this pattern (test that asynchronously reports completion on channel, deferred function that blocks until completion or timeout) in a lot of tests now, maybe we should refactor it into something that we can share across tests. future work